doc/source/whatsnew/v0.24.0.txt

```diff
@@ -879,7 +879,7 @@ Numeric
 Strings
 ^^^^^^^
 
--
+- Bug in :meth:`Series.str.cat` which falsely did not raise if `others` was a `set` (:issue:`23009`)
```

Comment:
this is an api breaking change, so move it there. make this declerative as well, e.g. Series.str.cat will now raise if others is a set